pkg/types/margin.go

```diff
@@ -8,16 +8,125 @@ type FuturesExchange interface {
+	CanWithdraw                 bool               `json:"canWithdraw"`
+	FeeTier                     int                `json:"feeTier"`
+	MaxWithdrawAmount           fixedpoint.Value             `json:"maxWithdrawAmount"`
+	Positions                   []FuturesUserPosition `json:"positions"`
+	TotalInitialMargin          fixedpoint.Value             `json:"totalInitialMargin"`
+	TotalMaintMargin            fixedpoint.Value             `json:"totalMaintMargin"`
+	TotalMarginBalance          fixedpoint.Value             `json:"totalMarginBalance"`
+	TotalOpenOrderInitialMargin fixedpoint.Value             `json:"totalOpenOrderInitialMargin"`
+	TotalPositionInitialMargin  fixedpoint.Value             `json:"totalPositionInitialMargin"`
+	TotalUnrealizedProfit       fixedpoint.Value             `json:"totalUnrealizedProfit"`
+	TotalWalletBalance          fixedpoint.Value             `json:"totalWalletBalance"`
+	UpdateTime                  int64              `json:"updateTime"`
 }
 
+// FuturesUserAsset define cross futures account asset
+type FuturesUserAsset struct {
+	Asset                  string `json:"asset"`
+	InitialMargin          fixedpoint.Value `json:"initialMargin"`
+	MaintMargin            fixedpoint.Value `json:"maintMargin"`
+	MarginBalance          fixedpoint.Value `json:"marginBalance"`
+	MaxWithdrawAmount      fixedpoint.Value `json:"maxWithdrawAmount"`
+	OpenOrderInitialMargin fixedpoint.Value `json:"openOrderInitialMargin"`
+	PositionInitialMargin  fixedpoint.Value `json:"positionInitialMargin"`
+	UnrealizedProfit       fixedpoint.Value `json:"unrealizedProfit"`
+	WalletBalance          fixedpoint.Value `json:"walletBalance"`
+}
+
+// FuturesUserPosition define cross futures account position
+type FuturesUserPosition struct {
+	Isolated               bool             `json:"isolated"`
+	Leverage               fixedpoint.Value           `json:"leverage"`
+	InitialMargin          fixedpoint.Value           `json:"initialMargin"`
+	MaintMargin            fixedpoint.Value           `json:"maintMargin"`
+	OpenOrderInitialMargin fixedpoint.Value           `json:"openOrderInitialMargin"`
+	PositionInitialMargin  fixedpoint.Value           `json:"positionInitialMargin"`
+	Symbol                 string           `json:"symbol"`
+	UnrealizedProfit       fixedpoint.Value           `json:"unrealizedProfit"`
+	EntryPrice             fixedpoint.Value           `json:"entryPrice"`
+	MaxNotional            fixedpoint.Value           `json:"maxNotional"`
+	PositionSide           string `json:"positionSide"`
+	PositionAmt            fixedpoint.Value           `json:"positionAmt"`
+	Notional               fixedpoint.Value           `json:"notional"`
+	IsolatedWallet         fixedpoint.Value           `json:"isolatedWallet"`
+	UpdateTime             int64            `json:"updateTime"`
+}
+
+// IsolatedFuturesAccount is for the isolated futures account
+type IsolatedFuturesAccount struct {
```

Comment:
what's the difference between IsolatedFuturesAccount and FuturesAccount,
I mean do they have same fields? if so, maybe you can use struct composition to reuse them?